Add missing string comparison type to equals comparisons

lahma opened the pull request:
` <AnalysisLevel>latest-Recommended</AnalysisLevel>` is a good starting point as it's basically industry best practices and items can be skipped when needed with `#pragma` with a good reason.

* Use `System.StringComparison.Ordinal` by default
* Fix minor other issues or disable with `#pragma`
* Configure inspection level and allow list in `Directory.Build.props`

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1309

Now can make separate PRs removing analysis allow list items and making things stricter after this.

lahma commented on src/OrchardCore/OrchardCore.Media.Core/Services/MediaSizeLimitAttribute.cs:
```diff
@@ -55,7 +55,7 @@ public void OnAuthorization(AuthorizationFilterContext context)
                 }
 
                 var effectiveRequestSizePolicy = context.FindEffectivePolicy<IRequestSizePolicy>();
-                if (effectiveRequestSizePolicy == null || effectiveRequestSizePolicy == this)
```
This equality cannot be true currently as `this` is not `IRequestSizePolicy`, seemed weird and caused analysis warning (as error)

MikeAlhayek commented:
@lahma I use to used `using System;` when not imported over using `System.StringComparison.Ordinal` otherwise the change looks good to me.

lahma commented:
I've added explicit using statements for `System` and removed the prefix from usages. I'll probably create another PR to enable implicit usings globally 😉
